Python/README.md

```diff
@@ -0,0 +1,156 @@
+
+def model_training():
+    print("model training")
+
+```
+
+### Naming variables
+
+Use snake_case for variable and function names e.g. example_variable
+
+- Be descriptive with the naming to enhance the readability and maintainability of the code
+- when storing the results of counts or percentages for example, they should be added at the end of the variable name e.g. disease_count, disease_percentage
+- Constants (variables that dont change value) should be defined at the start of a script. They should be written in all capital letters e.g. CONSTANT, VARIABLE
+
+```python
+
+# Constants should be created at start of script. Written in all capital letters
+MAX_STUDENTS_IN_CLASS = 30
+
+# Use snake_case when naming variables
+name_of_school = "Python school"
+
+# storing results of counts and percentages
+boys_in_class_percentage = 50
+boys_in_class_count = 15
+
+```
+
+### Consistency 
+
+- Use consistent naming conventions throughout the code you are writing e.g. snake_case throughout entire script.
+- When naming column names in data frames, use the same variable naming pattern as the rest of the code.
+
+```python
+# Standard library imports
+import os
+
+# importing third party libraries
+import pandas as pd
+
+# Importing application specifc / in house packages
+import my_package
+
+# Creating dictionary containing our example data
+data_dictionary = {
+    'example_column_1':[1,2,3],
+    'example_column_2':[4,5,6],
+    'example_column_3':[7,8,9]
+}
+
+# Converting dictionary into a pandas dataframe
+example_dataframe = pd.DataFrame(data_dictionary)
+```
+
+### Commenting 
+
+- Add comments to the code to explain what it does, unless the code is very self explanatory.
+- Include the reasoning behind your code, especially for more complicated pieces of code and functions. For functions in particular, comments should be added to explain what the function does and how it is used. For functions and classes, this is known as a docstring. An example is provided below. 
+
+```python
+# This is an example comment.
+
+def print_name(name,age):
+    """
+    Prints an individuals name and age
+
+    Args:
+        name (string): the individuals name
+        age (string): the individuals age
+
+    Returns:
+        sentence (string): A sentence that includes the name and age of the individual
+
+    """
+    sentence = f"My name is {}. I am {age} years old"
+    return(sentence)
+
+```
+
+
+### General programming principles
+
+- Keep it simple
+- Dont repeat yourself. Pieces of code should be implemented in just one place in the souce code. Tasks that are repeated should be implemented as functions
+- You are not going to need it. Do not implement something until it is necessary. 
+- For more complicated applications or pieces of software, separation of concerns principles should be implemented. Code should be as independent as possible, making it easier to maintain, update, and reuse. 
+
+
+
+## Notebook usage and best practice
```

Comment:
For Jupyter notebook, would be useful to add a specific note that outputs need to be cleared before committing and pushing to GitHub.

Reply:
A note has been added to this section under a new heading.